### PR TITLE
Add Hello XAI notebook

### PR DESCRIPTION
**Pull request by negvet**
Add basic OV XAI notebook

### CODE REVIEW

**Review by goodsong81 — COMMENTED** [review]
Looks all good to me :)
Besides, what do you think of changing the root directory name to "explainable-ai" so that we can add some detailed use-case notebooks in the directory afterwards?
Or, do you prefer to keep this simple and add another directory & readme for advanced ones?

**Comment by negvet** [comment]
> Looks all good to me :) Besides, what do you think of changing the root directory name to "explainable-ai" so that we can add some detailed use-case notebooks in the directory afterwards? Or, do you prefer to keep this simple and add another directory & readme for advanced ones?

As I can see, the pattern is the following: one notebook per directory.

So we probably will create a different directories for more advanced notebooks. I do not mind about renaming, I would propose to decide when we will define a scope of advanced use cases. For example I would prefer they to be located nearby: "explainable-ai-basic", "explainable-ai-advanced-a", "explainable-ai-advanced-b".

**Comment by goodsong81** [comment]
> > Looks all good to me :) Besides, what do you think of changing the root directory name to "explainable-ai" so that we can add some detailed use-case notebooks in the directory afterwards? Or, do you prefer to keep this simple and add another directory & readme for advanced ones?
> 
> As I can see, the pattern is the following: one notebook per directory.
> 
> So we probably will create a different directories for more advanced notebooks. I do not mind about renaming, I would propose to decide when we will define a scope of advanced use cases. For example I would prefer they to be located nearby: "explainable-ai-basic", "explainable-ai-advanced-a", "explainable-ai-advanced-b".

"nearby"! That was the only concern of mine :)
`hello-xai` directory might be shown a bit away from `explainable-ai-xxx`.

**Comment by GalyaZalesskaya** [comment]
I agree here with Evgeny to name XAI noteebooks to have them close to each other. So, this can be "explainable-ai-hello-world", "explainable-ai-basic"

**Comment by GalyaZalesskaya** [comment]
```
plt.imshow(explanation.saliency_map[result_index])
plt.title(f"Saliency map of the {result_index} class.")
```

I'd add here the name of predicted label in title together with class index.

**Comment by negvet** [comment]
> ```
> plt.imshow(explanation.saliency_map[result_index])
> plt.title(f"Saliency map of the {result_index} class.")
> ```
> 
> I'd add here the name of predicted label in title together with class index.

I also was trying to do that, but class name it too long: `n02099267 flat-coated retriever`. It is possible to process it, but I do not want to include this additional logic into the notebook. I have this class name in two cells around the final saliency map, so hopefully it is clear.

**Comment by GalyaZalesskaya** [comment]
Can you update notebook meta on the bottom:

```
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  },
  "openvino_notebooks": {
    "imageUrl": "https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/explainable-ai-2-deep-dive/retriever-saliency-map.jpg?raw=true", 
#add  here the picture with saliency map example from README
    "tags": {
        "categories": [
         "Explainable AI",
         "First Steps"
        ],
        "libraries": [],
        "other": [],
        "tasks": [
         "Image Classification",
         "Image-to-Image"
        ]
       }
   }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
```

**Comment by negvet** [comment]
@eaidova could you please help with CI issues? Not sure I have an idea of how to fix it. 

Some of the issues are other notebook related.

But there is also `ModuleNotFoundError: No module named 'openvino_xai'` in cell 2, although in cell 1 I have `%pip install -q  openvino-xai==1.0.0rc1`. Locally it is not reproduced.

**Comment by eaidova** [comment]
@negvet please try to wrap requirement into double quotas (some platforms requires such format for installing deps with version):
```
%pip install -q  "openvino-xai==1.0.0rc1"
```

**Comment by eaidova** [comment]
@negvet I tried your command in env with python3.8 and see such error:

```
ERROR: Ignored the following versions that require a different python version: 1.0.0rc0 Requires-Python >=3.10; 1.0.0rc1 Requires-Python >=3.10
ERROR: Could not find a version that satisfies the requirement openvino-xai (from versions: none)
ERROR: No matching distribution found for openvino-xai
```
it looks like package does not support python less than 3.10, that is why it can not pass on ci.

if it expected, then it should be noted in notebook and for specific python versions notebook should be added to ignored scope
https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/skipped_notebooks.yml
 (here example for ignoring for python3.8 https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/skipped_notebooks.yml#L517)

**Comment by negvet** [comment]
@eaidova thanks for your help!
I applied your fixes, but I am still getting

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[2], line 7
      5 import numpy as np
      6 import openvino as ov
----> 7 import openvino_xai as xai
      9 # Fetch `notebook_utils` module
     10 import requests

ModuleNotFoundError: No module named 'openvino_xai'
```
Please see:
https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/9875079299/job/27270906339?pr=2150

May be you have some other ideas of how to fix it?

**Comment by eaidova** [comment]
@negvet please also add notebook in this list https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/ignore_treon_docker.txt. Unfortunatly, this docker image uses python3.8, so the root cause for this issue is the same

File: notebooks/explainable-ai-1-basic/explainable-ai-1-basic.ipynb [hunk 1]
```diff
@@ -0,0 +1,456 @@
+{
```
Comment (eaidova) [inline notebooks/explainable-ai-1-basic/explainable-ai-1-basic.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    %pip install -q "openvino&gt;=2023.1.0" opencv-python tqdm
</pre><p>we prefer to use the latest version of openvino as basic point at the moment of notebook publication for notebook installation, please use "openvino&gt;=2024.2.0"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2150/discussion/#comment-1673462146'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

Comment (eaidova) [inline notebooks/explainable-ai-1-basic/explainable-ai-1-basic.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    %pip install -q  "openvino-xai==1.0.0rc1"
</pre><p>What is release schedule for openvino-xai? Generally, we avoid strict versions for installing packages for reduce maintenance overhead (when you need to update version for every new release), is it possible to use "openvino-xai&gt;=1.0.0"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2150/discussion/#comment-1673462150'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

Comment (eaidova) [inline notebooks/explainable-ai-1-basic/explainable-ai-1-basic.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    output_layer = compiled_model.output(0)
</pre><p>it is old syntax</p><p>there is no need to obtain output_layer, enough just use compiled_model([input_image])[0] in next cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2150/discussion/#comment-1673462152'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='9'/>

Comment (negvet) [reply notebooks/explainable-ai-1-basic/explainable-ai-1-basic.ipynb:1]: Fixed!

Comment (negvet) [reply notebooks/explainable-ai-1-basic/explainable-ai-1-basic.ipynb:1]: Sure, updated

**Comment by negvet** [comment]
> @negvet please also add notebook in this list https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/ignore_treon_docker.txt. Unfortunatly, this docker image uses python3.8, so the root cause for this issue is the same

I updated, seems better now, ready to be merged.
Thank you for help!

**Review by goodsong81 — APPROVED** [review]
LGTM, but looks like more treatment for a few CI failures.

**Comment by negvet** [comment]
@eaidova the rest issues are from other notebooks. Is it possible to merge like this or should I try to fix them?

**Comment by eaidova** [comment]
> @eaidova the rest issues are from other notebooks. Is it possible to merge like this or should I try to fix them?

yes, please ignore them (notebooks team is looking on that)

**Comment by eaidova** [comment]
@negvet  I have the last comment, could you please provide note about python version requirement in notebook (by default it is expected that openvino notebooks supports python 3.8-3.11, but your notebook is working only for python >= 3.10)?
you can find example of such note in this notebook https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/stable-diffusion-keras-cv/stable-diffusion-keras-cv.ipynb

File: notebooks/explainable-ai-1-basic/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# eXplainable AI (XAI) for OpenVINO™ IR Models
+[![Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/explainable-ai-basic/explainable-ai-basic.ipynb)
```
Comment (eaidova) [inline notebooks/explainable-ai-1-basic/README.md]: path for colab link is not matched with notebook path relative to repository root
```suggestion
[![Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/explainable-ai-1-basic/explainable-ai-1-basic.ipynb)
```

Comment (negvet) [reply notebooks/explainable-ai-1-basic/README.md]: good catch, thanks!

**Comment by negvet** [comment]
> @negvet I have the last comment, could you please provide note about python version requirement in notebook (by default it is expected that openvino notebooks supports python 3.8-3.11, but your notebook is working only for python >= 3.10)? you can find example of such note in this notebook https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/stable-diffusion-keras-cv/stable-diffusion-keras-cv.ipynb

Fixed it, thank you!

**Review by eaidova — APPROVED** [review]

**Comment by negvet** [comment]
@eaidova could you please help with two more questions:

1. How to enable XAI filter in the ecosystem?
image

2. How to change the notebook avatar? I want it to be overlay, not the original image
image

**Comment by eaidova** [comment]
@negvet regarding image, you need to update notebook metadata (open it in json mode and modify openvino_notebooks section)
now your notebook contains such info:

```
 "openvino_notebooks": {
   "imageUrl": "https://user-images.githubusercontent.com/36741649/127172572-1cdab941-df5f-42e2-a367-2b334a3db6d8.jpg",
   "tags": {
    "categories": [
     "First Steps"
    ],
    "libraries": [],
    "other": [],
    "tasks": [
     "Image Classification"
    ]
   }
  },
```
imageUrl is preview image for github pages.

Regading tag in ecosystem, @yatarkan could you please assist with addition OpenVINO XAI into dashbord (based on search by openvino_xai lib imports)?

additionally, I can sugges to add "XAI" tag into "other" category, then this category will be added to Quick filters and visible here:

image

**Pull request merged**